proposals/module-types/Example-SharedEverythingDynamicLinking.md

```diff
@@ -0,0 +1,525 @@
+# Shared-Everything Dynamic Linking Example
+
+This document walks through the 
+[Shared-Everything Dynamic Linking](Explainer.md#shared-everything-dynamic-linking)
+use case. One high-level point to make before starting is that the scheme
+outlined below avoids the need for any sort of runtime loader (like `ld.so`).
+Rather, Module Linking proposal is used in such a way that the wasm engine ends
+up doing everything.
+
+As with native, shared-everything dynamic linking is rather involved, requiring
+conventions followed by the toolchain, shared libraries and programs. We start
+with `libc`:
+
+## libc
+
+`libc` is special in that, conventionally, an implementation (like [wasi-libc])
+is bundled with a compiler (like clang) to produce an SDK (like [wasi-sdk]).
+
+For our hypothetical dynamic linking convention, we'll have `libc` define and
+export a linear memory which will be imported by the rest of the program, along
+with `malloc`, `free`, etc. Thus, `libc.wasm` is roughly shaped:
+
+```wasm
+;; libc.wasm
+(module
+  (memory (export "memory") 1)
```

Comment:
One thing missing here is the linear-memory stack pointer. With shared-everything linking, it would need to be shared as well, so that calls from one module to another can continue on the same linear-memory stack. Would it make sense to say that the program which defines and exports the memory to also defines and exports the mutable global for the stack pointer?

Reply:
Yep!  I'll add a note.